### PR TITLE
Systemtests olm upgrade

### PR DESCRIPTION
**Pull request by famartinrh**
### Type of change

- Systemtest

### Description

Includes two new test for olm upgrade process, adds the files for our custom-operator-registry used for the tests(which could be used in future to replace the olm-manifests image). Adds two new env variables, used in the process of build and deploy the custom operator registry.

This PR also includes two commits from the 0.31.0 release branch which are missing in master branch and are important in order for this new tests to run properly

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [x] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by famartinrh** [comment]
@enmasse-ci run tests profile=upgrade upgrade_from=0.30.2 ocp4 testcase=UpgradeTest#testUpgradeOLMSpecific

File: systemtests/custom-operator-registry/build_push_registry.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+FROM=${1}
+CUSTOM_IMAGE=${2}
+REGISTRY=${3}
+
+rm -rf manifests
+id=$(docker create $FROM)
+docker cp $id:/manifests $PWD
+if [ "${REGISTRY}" != "" ]; then
+    find manifests -name "*.yaml" | xargs sed -e "s,registry.redhat.io/amq7/amq-online-,${REGISTRY}/rh-osbs/amq7-amq-online-,g" -i
```
Comment (lulf) [inline systemtests/custom-operator-registry/build_push_registry.sh]: Should not refer to Red Hat product.

**Comment by famartinrh** [comment]
@enmasse-ci run tests profile=upgrade upgrade_from=0.30.2 ocp4

**Comment by famartinrh** [comment]
@enmasse-ci run tests profile=upgrade upgrade_from=0.30.2 ocp4

**Comment by famartinrh** [comment]
@enmasse-ci run tests profile=upgrade upgrade_from=0.30.2 ocp4

**Comment by famartinrh** [comment]
@enmasse-ci run tests profile=upgrade upgrade_from=0.30.2 ocp4

**Comment by famartinrh** [comment]
@enmasse-ci run tests profile=upgrade upgrade_from=0.29.2,0.30.2 ocp4

**Comment by famartinrh** [comment]
@enmasse-ci run tests profile=upgrade upgrade_from=0.29.2,0.30.2 ocp4

**Comment by famartinrh** [comment]
@enmasse-ci run tests profile=upgrade upgrade_from=0.30.2,0.31.0 ocp4

**Comment by famartinrh** [comment]
@enmasse-ci run tests profile=upgrade upgrade_from=0.30.2,0.31.0 ocp4

**Comment by famartinrh** [comment]
@enmasse-ci run tests profile=upgrade upgrade_from=0.30.2,0.31.0 ocp4

**Comment by famartinrh** [comment]
@enmasse-ci run tests profile=upgrade upgrade_from=0.30.2,0.31.0 ocp4

**Comment by lulf** [comment]
@famartinrh I think you need to rebase this

File: systemtests/custom-operator-registry/build_push_registry.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+FROM=${1}
+CUSTOM_IMAGE=${2}
+
+rm -rf manifests
+id=$(docker create $FROM)
```
Comment (lulf) [inline systemtests/custom-operator-registry/build_push_registry.sh:6]: Can you allow override the docker command using an env , I.e ${DOCKER}?

Comment (famartinrh) [reply systemtests/custom-operator-registry/build_push_registry.sh:6]: sure :)

File: systemtests/custom-operator-registry/build_push_registry.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+FROM=${1}
+CUSTOM_IMAGE=${2}
+
+CONTAINER_CTL=${CONTAINER_CTL:-docker}
```
Comment (kornys) [inline systemtests/custom-operator-registry/build_push_registry.sh:5]: Please use DOCKER env var like we use in all makefiles :) see Makefile.env.mk

Comment (famartinrh) [reply systemtests/custom-operator-registry/build_push_registry.sh]: done :)

**Pull request merged**